### PR TITLE
pyi cleanup

### PR DESCRIPTION
**Pull request by bdhirsh**
These are some followups from the first pyi codegen PR. Still maintaining byte-for-byte compatibility in this one.

- Separated `argument_str()` with a pyi flag into two functions, `argument_str()` and `argument_str_pyi()`
- Added a notes section for pyi at the top of `python.py`
- Added a `Python Interface` section that I moved the free-standing pyi functions to

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49057 pyi codegen refactor - no need to group python signatures by overload name
* #49056 pyi codegen - removing byte-for-byte-compatibility hacks (sorting overloads)
* #49055 pyi codegen - removing byte-for-byte compatibility hacks
* **#49054 pyi cleanup**

Differential Revision: [D25410848](https://our.internmc.facebook.com/intern/diff/D25410848)

### CODE REVIEW

File: .jenkins/pytorch/codegen-test.sh [hunk 1]
```diff
@@ -37,7 +37,6 @@ python -m tools.setup_helpers.generate_code \
 mkdir -p "$OUT"/pyi/torch/_C
 mkdir -p "$OUT"/pyi/torch/nn
 python -m tools.pyi.gen_pyi \
-  --declarations-path "$OUT"/torch/share/ATen/Declarations.yaml \
```
Comment (bdhirsh) [inline .jenkins/pytorch/codegen-test.sh:40]: I should have taken this out in the original PR

File: tools/autograd/gen_python_functions.py [hunk 2]
```diff
@@ -228,8 +228,6 @@ def signature_original(f: NativeFunction) -> str:
         opname = str(f.func.name.name.base)
         if f.func.is_out_fn():
             opname += '_out'
-        # TODO: remove HACK
-        # I think we want to differentiate inplace functions here.. but we currently don't for the arg parser
```
Comment (bdhirsh) [inline tools/autograd/gen_python_functions.py:232]: As Ed pointed out, this is actually the right behavior for the python arg parser.

File: .jenkins/pytorch/codegen-test.sh (hunk 1 above)
Comment (ljk53) [reply .jenkins/pytorch/codegen-test.sh:40]: I should have integrated this script with CI properly...

**Review by ljk53 — APPROVED** [review]
Thanks!

**Pull request closed without merging**